narwhals/stable/v1/__init__.py

```diff
@@ -738,7 +739,7 @@ def from_native(
 
 @overload
 def from_native(
-    native_dataframe: Any,
+    native_dataframe: IntoSeriesT | Any,  # remain `Any` for downstream compatibility
```

Comment:
for https://github.com/vega/altair/blob/796649efd751a7b928ac4f81e192d0ae93f4a3d0/altair/utils/schemapi.py#L502C1-L507C25

```python
def _from_array_like(obj: Iterable[Any], /) -> list[Any]:
    try:
        ser = nw.from_native(obj, strict=True, series_only=True)
        return ser.to_list()
    except TypeError:
        return list(obj)
```